### DISCUSSION TITLE
how can i get the StartupEventArgs

### DISCUSSION THREAD

**Discussion by zombers** [PrismLibrary/Prism#2574] [WPF]
i have try to override the method "OnStartup",but when the StartupEventArgs.arg.length bigger than 1, my application will freeze(just can see a white page)

**Comment by zombers** [comment]
我需要双击某类型文件时，直接打开我的应用。但是我override  OnStartup方法之后，只要StartupEventArgs.arg有内容 我的程序就会只弹出来一个白色页面。我不知道他卡死在哪一步了。
代码如下:
protected override void OnStartup(StartupEventArgs e)
{
if (e.Args.Length > 0){//}
base.OnStartup(e);
OnInitialized();
}

**Reply by brianlagunas** [reply]
Hard to say without seeing your code. Chances are you are doing something that causes the app to not work properly. A reproduction app would go a long way in helping find the issue.

**Reply by zombers** [reply]
in App.xaml.cs ,I override the method "OnStartup".
when the parameter StartupEventArgs.arg length more than 0.
my application will just show me a white page.
code:
protected override void OnStartup(StartupEventArgs e)
{
if (e.Args.Length > 0){//}
base.OnStartup(e);
OnInitialized();
}

**Reply by brianlagunas** [reply]
Reposting what you already posted does not help. How about a sample app the demonstrates the problem?

**Comment by zombers** [comment]
![pp](https://user-images.githubusercontent.com/26043286/135369254-19d3ef49-9576-4373-b7e4-e7e0199315b2.gif)

**Reply by brianlagunas** [reply]
That is not helpful.

**Reply by zombers** [reply]
when i run the application ,it will work success(StartupEventArgs.arg no parameter).But when i try to  use *.edf file to open the application,i just can see a white application(StartupEventArgs.arg  parameter is the file path)

**Reply by brianlagunas** [reply]
Then you have a problem in your code. No one can help you without the ability to run the offending code. What have you tried doing to fix it?

**Reply by zombers** [reply]
i just want to no how can i get the StartupEventArgs  parameter.without the method "OnStartup"

**Reply by brianlagunas** [reply]
You don't. You have to find the issue in your code and fix it.

**Reply by zombers** [reply]
i konw U mean. 
when exegesis the method,i can run it two way .
It seems that this method comes out of the problem


protected override void OnStartup(StartupEventArgs e)
{
  if (e.Args.Length > 0)
  {//save the path}
  base.OnStartup(e);
  OnInitialized();
}

**Reply by zombers** [reply]
I don't know the real use of this method

**Reply by brianlagunas** [reply]
I'm sorry, but you haven't provided any information, code, or context that anyone can use to solve your problem. The best guess I can even give is to move the call to 'base.OnStartUp()' first before the rest of the code

**Reply by zombers** [reply]
I'm so sorry. My English level is a little poor, which affects our communication very much. I'll give you a simplified app.xaml .cs

**Comment by zombers** [comment]
I'm sorry, but I've temporarily stopped solving this problem for work reasons. The last time I re-debugged, I found that my code wasn't working properly for other reasons, and I'll contact you when I'm free.
从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

________________________________
发件人: Brian Lagunas ***@***.***>
发送时间: Thursday, September 30, 2021 9:34:06 AM
收件人: PrismLibrary/Prism ***@***.***>
抄送: zombers ***@***.***>; Author ***@***.***>
主题: Re: [PrismLibrary/Prism] how can i get the StartupEventArgs (Discussion #2574)


I'm sorry, but you haven't provided any information, code, or context that anyone can use to solve your problem. The best guess I can even give it to move the call to 'base.OnStartUp()' first before the rest of the code

―
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/PrismLibrary/Prism/discussions/2574#discussioncomment-1406195>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGGWHFQWK2VCZK4BCO2BEL3UEO5A5ANCNFSM5E7RNXWA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
